packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,247 @@
+			}
+		}
+
+		async function onSetLatestPostsHomepage( event: React.FormEvent ) {
+			event.preventDefault();
+
+			try {
+				await editEntityRecord( 'root', 'site', undefined, {
+					page_for_posts: 0,
+					page_on_front: 0,
+					show_on_front: 'posts',
+				} );
+
+				closeModal?.();
+
+				await saveEditedEntityRecord( 'root', 'site', undefined, {
+					page_for_posts: 0,
+					page_on_front: 0,
+					show_on_front: 'posts',
+				} );
+
+				createSuccessNotice(
+					__( 'Homepage set to display latest posts' ),
+					{
+						type: 'snackbar',
+					}
+				);
+				onActionPerformed?.( items );
+			} catch ( error ) {
+				const typedError = error as CoreDataError;
+				const errorMessage =
+					typedError.message && typedError.code !== 'unknown_error'
+						? typedError.message
+						: __(
+								'An error occurred while setting the homepage to display latest posts'
+						  );
+				createErrorNotice( errorMessage, { type: 'snackbar' } );
+			}
+		}
+
+		const renderModalBody = () => {
+			if ( 'posts' === showOnFront ) {
+				return (
+					<>
+						<Text>
+							{ sprintf(
+								// translators: %s: title of the page to be set as the homepage.
+								__(
+									'Set "%s" as the site homepage? This will replace the current homepage which is set to display latest posts.'
+								),
+								pageTitle
+							) }
+						</Text>
+					</>
+				);
+			}
+
+			const modalTranslatedString =
+				// translators: %1$s: title of page to be set as the home page. %2$s: title of the current home page.
+				__(
+					'Set "%1$s" as the site homepage? This will replace the current homepage: "%2$s"'
+				);
+
+			return (
+				<Text>
+					{ sprintf(
+						modalTranslatedString,
+						pageTitle,
+						currentHomePageTitle
+					) }{ ' ' }
+				</Text>
+			);
+		};
+
+		const submitAction =
+			showOnFront === 'posts' || item.id !== pageForPosts
+				? onSetPageAsHomepage
+				: onSetLatestPostsHomepage;
+
+		return (
+			<form onSubmit={ submitAction }>
+				<VStack spacing="5">
+					{ renderModalBody() }
+					<HStack justify="right">
+						<Button
+							__next40pxDefaultSize
+							variant="tertiary"
+							onClick={ () => {
+								closeModal?.();
+							} }
+						>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							__next40pxDefaultSize
+							variant="primary"
+							type="submit"
+						>
+							{
+								// translators: Button to confirm setting the specified page as the homepage.
+								__( 'Set homepage' )
+							}
+						</Button>
+					</HStack>
+				</VStack>
+			</form>
+		);
+	};
+
+const setAsHomepage: Action< PostWithPermissions > = {
+	id: 'set-as-homepage',
+	label: __( 'Set as homepage' ),
+	isEligible( post ) {
+		if ( post.status === 'trash' ) {
+			return false;
+		}
+
+		if ( post.type !== PAGE_POST_TYPE ) {
+			return false;
+		}
+
+		// A front-page template overrides homepage settings, so don't show the action if it's present.
```

Comment:
Can you expand on this? I don't think the setting has anything to do with templates, and the setting is honored, no?

Reply:
If a theme includes a `front-page.html` template, this will override any homepage setting, so we shouldn't show the "Set as homepage" action as it will have no effect on the homepage settings.

I tried removing these lines to see if the setting was honored, but it looks like this check is required. I've refactored it so that the `useSelect` is not in `isEligible` anymore: https://github.com/WordPress/gutenberg/pull/65426/commits/3c715337516392d850fee35dad2bc753c0159c20.

Reply:
This check doesn't depend on the item. It's a global check like any other we have in `packages/editor/src/dataviews/store/private-actions.ts`. Can we move it there instead? There's a performance advantage to do so: the action won't be registered at all.

Reply:
Yeah, this makes more sense, thank you. Moved `hasFrontPageTemplate ` to private actions here: https://github.com/WordPress/gutenberg/pull/65426/commits/24dcf8f583feeac099df7dae5162ca47bfad0f2f.